beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1725,13 +1733,63 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         result
     }
 
+    fn prune_abandoned_forks(
+        &self,
+        old_finalized_block: &Hash256,
+        new_finalized_block: &Hash256,
+    ) -> Result<(), Error> {
+        let old_finalized_slot = self
+            .get_block(old_finalized_block)?
+            .ok_or_else(|| Error::MissingBeaconState(*old_finalized_block))?
+            .message
+            .slot;
+
+        let new_finalized_slot = self
```

Comment:
Relating to the above comment, it's safest to do what you have done here instead of using `old_finalized_epoch.start_slot(..)`. This will ensure we prune the most blocks.

However, `new_finalized_slot` is known to `after_finalization`, so perhaps we could pass it in?

Reply:
Sure thing

Reply:
Considering that I got this mixed-up (thanks @michaelsproul for catching that), I think we should use `new_finalized_epoch.start_slot(..)` here.